src/web/js/google-apis/picker.js

```diff
@@ -0,0 +1,128 @@
+  console.log("Initializing File Picker");
+  // this.openOn does not need initialization
+};
+
+FilePicker.prototype.initOpen = function(drive, picker) {
+
+  var validateSelection = (function(sel) {
+    if (sel.length === 0) {
+      // Should be impossible, since we only run this if the user
+      // has picked something (see pickerCallback)
+      this.onInternalError("Internal Error (please report to developers): "
+                           + "Should be impossible: "
+                           + "Picker selection length is zero");
+    } else {
+      var firstType = sel[0][picker.Document.TYPE];
+      // Check that all selected files are of the same type
+      sel.forEach((function(f) {
+        if (f[picker.Document.TYPE] !== firstType) {
+          this.onError("Cannot open files of multiple types "
+                       + "(Expected: " + firstType
+                       + "; Received: " + f[picker.Document.TYPE]
+                       + ")");
+        }
+      }).bind(this));
+      switch (sel[0][picker.Document.TYPE]) {
+      case "file": // Pyret file
+        break;
+      case picker.Type.PHOTO: // Photo
+        // Fix photo permissions
+        var permissions = {
+          'role': 'reader',
+          'type': 'anyone',
+          'value': 'default',
+          'withLink': true
+        };
+        sel.forEach((function(photo) {
+          var fileId = photo[picker.Document.ID];
+          gwrap.drive.permissions.insert({
+            'fileId': fileId,
+            'resource': permissions
+          }).catch(this.onInternalError);
+        }).bind(this));
+        break;
+      default:
+        this.onError("Cannot open file of unknown type: " + sel[0][picker.Document.TYPE]);
+      }
+    }
+  }).bind(this);
+
+  var pickerCallback = (function(data) {
+    if (data[picker.Response.ACTION] == picker.Action.PICKED) {
+      var selection = data[picker.Response.DOCUMENTS];
+      validateSelection(selection);
+      this.dataHandler(selection, picker, drive);
+    }
+  }).bind(this);
+
+  var showPicker = (function() {
+    /**
+     * A Picker View which displays Pyret files which users may load.
+     */
+    var pyretView = new picker.DocsView(picker.ViewId.DOCUMENTS);
+
+    // [Image picker code is based on the Picker code from WeScheme:
+    // https://github.com/bootstrapworld/wescheme2012/blob/5b03bac247ac182c356d73fc3e7283ef1086777f/war-src/js/openEditor/editor.js#L571-L659 ]
+    /**
+     * A Picker View which displays images users may load.
+     */
+    var imageView = new picker.View(picker.ViewId.DOCS);
+    imageView.setMimeTypes("image/png,image/jpeg,image/jpg,image/gif");
+
+    var buildInstance = (function(parentId) {
+      pyretView.setParent(parentId);
+
+      this.pickerInstance = new picker.PickerBuilder()
+        //.enableFeature(picker.Feature.NAV_HIDDEN)
+        .enableFeature(picker.Feature.MULTISELECT_ENABLED)
+        .setTitle("Select a Pyret File or Image")
```

Comment:
Many of the messages we've presented to students before have said "Try doing this thing or that thing to fix problem X".  But since you're using this in the context of either picking files or picking images, seems like you should be able to pass in the string to specialize the prompt, so that it's clearer what action should be taken right now.